lib/spack/spack/compiler.py

```diff
@@ -311,11 +311,14 @@ def __init__(
         # Unfortunately have to make sure these params are accepted
         # in the same order they are returned by sorted(flags)
         # in compilers/__init__.py
-        self.flags = {}
-        for flag in spack.spec.FlagMap.valid_compiler_flags():
+        self.flags = spack.spec.FlagMap(self.spec)
+        for flag in self.flags.valid_compiler_flags():
+            # TODO: This should be a FlagMap not dict
```

Comment:
Is this comment still relevant?

Reply:
@tldahlgren The comment is in reference to `kwargs`. I'll modify the comment to make that more explicit. But this should be fixed at some point, even if it's not in this PR